src/coreclr/src/jit/emit.cpp

```diff
@@ -5880,19 +5942,33 @@ void emitter::emitOutputDataSec(dataSecDsc* sec, BYTE* dst)
 #ifdef DEBUG
             if (EMITVERBOSE)
             {
-                printf("  ");
+                printf("  section %3u, size %2u, RWD%2u:\t", secNum++, dscSize, curOffs);
+
                 for (size_t i = 0; i < dscSize; i++)
                 {
                     printf("%02x ", dsc->dsCont[i]);
                     if ((((i + 1) % 16) == 0) && (i + 1 != dscSize))
                     {
-                        printf("\n  ");
+                        printf("\n\t\t\t\t\t");
                     }
                 }
+                switch (dsc->dsDataType)
+                {
+                    case TYP_FLOAT:
+                        printf(" ; float  %9.6g", (double)*reinterpret_cast<float*>(&dsc->dsCont));
+                        break;
+                    case TYP_DOUBLE:
+                        printf(" ; double %12.9g", *reinterpret_cast<double*>(&dsc->dsCont));
```

Comment:
We already have printed the binary value. 
Additionally we print the floating point value as a comment in the Asm output to help the user see what value is being loaded.

Reply:
👍, I saw us printing the binary value elsewhere but wasn't sure where in the dump that was relative to this.

It would still be nice if we could ensure that values don't conflict in other places though (IMO).
For example, trimming `double` to 9 digits will result in both of the following looking equal:
* `3.141592653589793 (0x400921FB54442D18)` -- Math.PI
* `3.14159265               (0x400921FB53C8D4F1)`

However, they have 8 million uniquely representable values that exist between them.